Patch 1

axelvonkamp opened the pull request:
Basic implementation of multi-threaded FVA via an external JAVA virtual machnine with CPLEX as solver.

**I hereby confirm that I have:**

- [ X] Tested my code on my own machine
- [X ] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X ] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

tpfau commented:
Whats the advantage of this java implementation vs using `fluxVariability`which is already parallelized?
Incidentially, if CPLEX is only run on one thread, you might actually be slower on some problems, as this will only run the simplex method (see [here](https://www.ibm.com/support/knowledgecenter/en/SS9UKU_12.4.0/com.ibm.cplex.zos.help/UsrMan/topics/cont_optim/simplex/05_optimizer_auto_seln.html))

axelvonkamp commented:
Hi,

this implementation can use the warm-start capability of CPLEX and therefore is comparable to fastFVA. It also does not require the Matlab parallel toolbox.

-Axel

-- 
Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.Am 20.02.19, 13:01, Thomas Pfau <notifications@github.com> schrieb:
Whats the advantage of this java implementation vs using fluxVariabilitywhich is already parallelized?
Incidentially, if CPLEX is only run on one thread, you might actually be slower on some problems, as this will only run the simplex method (see here)

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/opencobra/cobratoolbox","title":"opencobra/cobratoolbox","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/opencobra/cobratoolbox"}},"updates":{"snippets":[{"icon":"PERSON","message":"@tpfau in #1427: Whats the advantage of this java implementation vs using `fluxVariability`which is already parallelized?\r\nIncidentially, if CPLEX is only run on one thread, you might actually be slower on some problems, as this will only run the simplex method (see [here](https://www.ibm.com/support/knowledgecenter/en/SS9UKU_12.4.0/com.ibm.cplex.zos.help/UsrMan/topics/cont_optim/simplex/05_optimizer_auto_seln.html))"}],"action":{"name":"View Pull Request","url":"https://github.com/opencobra/cobratoolbox/pull/1427#issuecomment-465546141"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/opencobra/cobratoolbox/pull/1427#issuecomment-465546141",
"url": "https://github.com/opencobra/cobratoolbox/pull/1427#issuecomment-465546141",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

laurentheirendt commented:
@axelvonkamp, may you make the requested changes? Thanks.

axelvonkamp commented:
> 
> 
> Please can you create an option in fluxVariability.m that allows one to use this nice contribution?

I am not sure if this can be easily done because fluxVariability.m already has many options and supporting all the different 'method' variants as well as the allowLoops parameter will require a lot of additional code. I think integrating it for method = 'FBA' and allowLoops = true should be relatively easy and can look into this at the end of next week.

axelvonkamp commented:
I have now added a new option to fluxVariability.m so that (the modified) mtFVA is now called from fluxVariability.

laurentheirendt commented:
May you add a test here? This is quite important. Testing guidelines are [here](https://opencobra.github.io/cobratoolbox/stable/contributing.html#guide-for-writing-a-test).

axelvonkamp commented:
> 
> 
> May you add a test here? This is quite important. Testing guidelines are [here](https://opencobra.github.io/cobratoolbox/stable/contributing.html#guide-for-writing-a-test).

I have added a new test in the test/verifiedTests/analysis/testFVA directory, is this the right place?
I have also found a bug in fluxVariability: the fluxes in minFlux, maxFlux might not be in the right order when using the rxnNameList parameter; this bug was masked by another bug in testFVA that I also fixed.

axelvonkamp commented:
Which Java compiler (javac) is installed on the Linux machines?

laurentheirendt commented:
```
$ javac -version
javac 1.8.0_201
```
There was none installed before - I installed `javac` just now.
